src/server/plugins/engine/plugin.ts

```diff
@@ -80,9 +80,12 @@ export const plugin = {
     const forms = server.app.forms
 
     configs.forEach((config) => {
+      const basePath =
+        formBasePath === '' ? config.id : `${formBasePath}/${config.id}`
```

Comment:
Could we use POSIX `join()` here to always do forward slashes?

```patch
-import { join } from 'node:path'
+import { join } from 'node:path/posix'
```

I think it skips the `/` when one of the path segments is empty

```suggestion
      const basePath = join(formBasePath, config.id)
```

Reply:
Agree we probably shouldn't be string concating URLs, but I'm not sure we should be using `node:path` for URLs, even the `posix` variant.

There are other (file path) uses of the imported `join` in that file too, which I wouldn't want to break for Windows users.

Reply:
Can add a whole new import for `posix.join()`?

Seemed like a nice way to handle **appPathPrefix** as `''` but it also works with  `'/'`

```patch
import { join } from 'node:path'
+import posix from 'node:path/posix'
```

```suggestion
      const basePath = posix.join(formBasePath, config.id)
```

Reply:
Your note about `''` and `'/'` made me think again about this. Initially I wanted to default `appPathPrefix` to undefined but there were assumptions in the code that it would be a string. I have removed those assumptions now and introduced an `appPathPrefixer` function that is now used in `js` and `njk` files. I've gone with `posix.join` for now (but don't think it should be used for building urls generally). I have also applied a joi schema to the `appPathPrefix` configuration key to assert it is a valid prefix (it now must be a valid relative uri, empty string and `'/'` are disallowed)